spec/system/page_objects/pages/user_preferences_security.rb

```diff
@@ -10,9 +10,10 @@ def visit(user)
 
       def visit_second_factor(password)
         click_button "Manage Two-Factor Authentication"
-
-        find(".dialog-body input#password").fill_in(with: password)
-        find(".dialog-body .btn-primary").click
+        find(".confirm-session input#password").fill_in(with: password)
+        find(".confirm-session .btn-primary:not([disabled])").click
+        has_no_css?(".confirm-session")
```

Comment:
We expect the dialog to be closed after the button has been clicked and the request is processed.